test/extensions/filters/http/alternate_protocols_cache/filter_integration_test.cc

```diff
@@ -144,6 +160,125 @@ TEST_P(FilterIntegrationTest, AltSvc) {
   test_server_->waitForCounterEq("cluster.cluster_0.upstream_cx_http3_total", 1);
 }
 
+TEST_P(FilterIntegrationTest, AltSvcCached) {
+  // Start with the alt-svc header in the cache.
+  fill_cache_ = true;
+
+  const uint64_t request_size = 0;
+  const uint64_t response_size = 0;
+  const std::chrono::milliseconds timeout = TestUtility::DefaultTimeout;
+
+  initialize();
+  codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
+
+  Http::TestRequestHeaderMapImpl request_headers{
+      {":method", "POST"},       {":path", "/test/long/url"},
+      {":scheme", "http"},       {":authority", "sni.lyft.com"},
+      {"x-lyft-user-id", "123"}, {"x-forwarded-for", "10.0.0.1"}};
+  int port = fake_upstreams_[1]->localAddress()->ip()->port();
+  std::string alt_svc = absl::StrCat("h3=\":", port, "\"; ma=86400");
+  Http::TestResponseHeaderMapImpl response_headers{{":status", "200"}, {"alt-svc", alt_svc}};
+
+  // First request should go out over HTTP/3 (upstream index 1) because of the Alt-Svc information.
+  auto response2 = sendRequestAndWaitForResponse(request_headers, request_size, response_headers,
+                                                 response_size, 1, timeout);
+  checkSimpleRequestSuccess(request_size, response_size, response2.get());
+  test_server_->waitForCounterEq("cluster.cluster_0.upstream_cx_http3_total", 1);
+}
+
+TEST_P(FilterIntegrationTest, AltSvcCachedH3Slow) {
+  // Start with the alt-svc header in the cache.
+  fill_cache_ = true;
+
+  const uint64_t request_size = 0;
+  const uint64_t response_size = 0;
+  const std::chrono::milliseconds timeout = TestUtility::DefaultTimeout;
+
+  initialize();
+  codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
```

Comment:
nit: is there an extra set of ()s around `lookupPort`? Here and elsewhere.

Reply:
yep!